exporter/collector/googlemanagedprometheus/monitoredresource.go

```diff
@@ -52,6 +53,27 @@ func MapToPrometheusTarget(res pcommon.Resource) *monitoredrespb.MonitoredResour
 	}
 }
 
+// AddTargetInfo extracts the target_info metric from each ResourceMetric associated with the input pmetric.Metrics
+// and inserts it into each ScopeMetric for that resource, with the matching "job" and "instance" labels as specified in
+// https://github.com/open-telemetry/opentelemetry-specification/blob/v1.16.0/specification/compatibility/prometheus_and_openmetrics.md#resource-attributes-1
+func AddTargetInfo(m pmetric.Metrics) {
```

Comment:
can you put this in its own file?  This one is called monitoredresource.go, and this probably doesn't fit there.

Reply:
Sure, I just put it here because it uses the gmp MR so it felt like it fit